File: src/lib/battle/equipment-templates.ts
/**
 * Equipment Templates and Slot Configuration
 *
 * This module contains all predefined equipment templates and
 * slot configurations by unit class. Extracted from equipment-system.ts
 * to reduce file size and improve maintainability.
 */

import type { UnitClass } from './advanced-unit'
import type {
  Equipment,
  EquipmentSlot,
  EquipmentAbility,
  AbilityEffect,
  TriggerCondition,
} from './equipment-system'

// ============================================================================
// EQUIPMENT SLOTS CONFIGURATION
// ============================================================================

/**
 * Default slot configuration by unit class
 */
export const DEFAULT_SLOT_CONFIG: Record<UnitClass, Record<EquipmentSlot, number>> = {
  fighter: {
    weapon: 1,
    secondary: 0,
    shield: 1,
    armor: 1,
    engine: 1,
    computer: 0,
    special: 0,
    consumable: 1,
  },
  corvette: {
    weapon: 1,
    secondary: 1,
    shield: 1,
    armor: 1,
    engine: 1,
    computer: 1,
    special: 0,
    consumable: 1,
  },
  frigate: {
    weapon: 2,
    secondary: 1,
    shield: 1,
    armor: 1,
    engine: 1,
    computer: 1,
    special: 1,
    consumable: 2,
  },
  cruiser: {
    weapon: 2,
    secondary: 2,
    shield: 2,
    armor: 2,
    engine: 1,
    computer: 1,
    special: 1,
    consumable: 2,
  },
  battlecruiser: {
    weapon: 3,
    secondary: 2,
    shield: 2,
    armor: 2,
    engine: 1,
    computer: 2,
    special: 1,
    consumable: 2,
  },
  battleship: {
    weapon: 4,
    secondary: 2,
    shield: 2,
    armor: 3,
    engine: 1,
    computer: 2,
    special: 2,
    consumable: 3,
  },
  dreadnought: {
    weapon: 5,
    secondary: 3,
    shield: 3,
    armor: 3,
    engine: 1,
    computer: 2,
    special: 3,
    consumable: 3,
  },
  carrier: {
    weapon: 2,
    secondary: 2,
    shield: 2,
    armor: 2,
    engine: 1,
    computer: 2,
    special: 4,
    consumable: 2,
  },
  transport: {
    weapon: 0,
    secondary: 1,
    shield: 2,
    armor: 2,
    engine: 2,
    computer: 1,
    special: 1,
    consumable: 2,
  },
  utility: {
    weapon: 0,
    secondary: 1,
    shield: 1,
    armor: 1,
    engine: 2,
    computer: 2,
    special: 1,
    consumable: 0,
  },
  // Defense structures
  platform: {
    weapon: 2,
    secondary: 1,
    shield: 2,
    armor: 2,
    engine: 0,
    computer: 1,
    special: 1,
    consumable: 1,
  },
  turret: {
    weapon: 1,
    secondary: 0,
    shield: 1,
    armor: 1,
    engine: 0,
    computer: 1,
    special: 0,
    consumable: 0,
  },
  defense: {
    weapon: 2,
    secondary: 1,
    shield: 2,
    armor: 3,
    engine: 0,
    computer: 1,
    special: 1,
    consumable: 0,
  },
  missile: {
    weapon: 3,
    secondary: 0,
    shield: 1,
    armor: 1,
    engine: 0,
    computer: 2,
    special: 1,
    consumable: 0,
  },
}

// ============================================================================
// PREDEFINED EQUIPMENT TEMPLATES
// ============================================================================

/**
 * Common equipment templates
 * These can be instantiated using createEquipment() from equipment-system.ts
 */
export const EQUIPMENT_TEMPLATES: Record<string, Omit<Equipment, 'id'>> = {
  // ========== WEAPONS ==========
  gauss_cannon: {
    name: 'Gauss Cannon',
    description: 'Electromagnetic railgun with high armor penetration.',
    slot: 'weapon',
    rarity: 'common',
    activationType: 'passive',
    compatibleClasses: ['frigate', 'cruiser', 'battlecruiser', 'battleship', 'dreadnought'],
    statModifiers: {
      damageTypeBonuses: { ballistic: 50 },
      accuracyBonus: 5,
    },
    abilities: [],
    triggers: [],
    requirements: {},
    installCost: { metal: 5000, crystal: 2000, deuterium: 500 },
    powerConsumption: 10,
    mass: 100,
  },

  ion_disruptor: {
    name: 'Ion Disruptor',
    description: 'Specialized weapon that disrupts enemy shields and systems.',
    slot: 'weapon',
    rarity: 'uncommon',
    activationType: 'passive',
    compatibleClasses: ['cruiser', 'battlecruiser', 'battleship', 'dreadnought'],
    statModifiers: {
      damageTypeBonuses: { ionic: 80 },
    },
    abilities: [],
    triggers: [
      {
        condition: { trigger: 'on_hit', chance: 20 },
        effect: {
          type: 'debuff',
          target: 'enemy',
          statusEffect: 'shield_disruption',
          duration: 2,
          strength: 50,
        } as AbilityEffect,
      },
    ],
    requirements: { techLevels: { weaponsTech: 8 } },
    installCost: { metal: 8000, crystal: 6000, deuterium: 1000 },
    powerConsumption: 15,
    mass: 80,
  },

  plasma_lance: {
    name: 'Plasma Lance',
    description: 'High-energy plasma weapon with devastating critical damage.',
    slot: 'weapon',
    rarity: 'rare',
    activationType: 'passive',
    compatibleClasses: ['battleship', 'dreadnought'],
    statModifiers: {
      damageTypeBonuses: { explosive: 100, ballistic: 50 },
      critChanceBonus: 10,
      critDamageBonus: 0.5,
    },
    abilities: [],
    triggers: [],
    requirements: { techLevels: { weaponsTech: 12 } },
    installCost: { metal: 15000, crystal: 10000, deuterium: 3000 },
    powerConsumption: 25,
    mass: 150,
  },

  torpedo_launcher: {
    name: 'Torpedo Launcher',
    description: 'Launches guided torpedoes with high explosive damage.',
    slot: 'secondary',
    rarity: 'common',
    activationType: 'passive',
    compatibleClasses: ['corvette', 'frigate', 'cruiser', 'battlecruiser', 'battleship', 'dreadnought'],
    statModifiers: {
      damageTypeBonuses: { explosive: 75 },
    },
    abilities: [
      {
        id: 'salvo',
        name: 'Torpedo Salvo',
        description: 'Fire a concentrated salvo at a single target.',
        cooldown: 3,
        currentCooldown: 0,
        energyCost: 20,
        duration: 0,
        effect: {
          type: 'damage',
          damage: { explosive: 200 },
          targetCount: 1,
          areaEffect: false,
          penetration: 20,
        } as AbilityEffect,
      },
    ],
    triggers: [],
    requirements: {},
    installCost: { metal: 4000, crystal: 2000, deuterium: 1000 },
    powerConsumption: 8,
    mass: 60,
  },

  // ========== SHIELDS ==========
  reinforced_shields: {
    name: 'Reinforced Shield Generator',
    description: 'Enhanced shield generator with improved capacity.',
    slot: 'shield',
    rarity: 'common',
    activationType: 'passive',
    compatibleClasses: ['fighter', 'corvette', 'frigate', 'cruiser', 'battlecruiser', 'battleship', 'dreadnought', 'carrier', 'transport', 'utility'],
    statModifiers: {
      shieldMultiplier: 1.2,
      shieldRegenBonus: 5,
    },
    abilities: [],
    triggers: [],
    requirements: {},
    installCost: { metal: 3000, crystal: 4000, deuterium: 500 },
    powerConsumption: 10,
    mass: 50,
  },

  adaptive_shields: {
    name: 'Adaptive Shield Matrix',
    description: 'Shields that adapt to incoming damage types.',
    slot: 'shield',
    rarity: 'rare',
    activationType: 'passive',
    compatibleClasses: ['cruiser', 'battlecruiser', 'battleship', 'dreadnought', 'carrier'],
    statModifiers: {
      shieldMultiplier: 1.15,
      resistanceBonuses: {
        ballistic_resistance: 10,
        ionic_resistance: 10,
        explosive_resistance: 10,
      },
    },
    abilities: [],
    triggers: [
      {
        condition: { trigger: 'on_damage', chance: 100 },
        effect: {
          type: 'buff',
          target: 'self',
          modifiers: {
            resistanceBonuses: {
              ballistic_resistance: 5,
              ionic_resistance: 5,
              explosive_resistance: 5,
            },
          },
          duration: 1,
        } as AbilityEffect,
      },
    ],
    requirements: { techLevels: { shieldTech: 10 } },
    installCost: { metal: 10000, crystal: 15000, deuterium: 2000 },
    powerConsumption: 20,
    mass: 80,
  },

  emergency_shields: {
    name: 'Emergency Shield Booster',
    description: 'Activatable shield boost for critical moments.',
    slot: 'shield',
    rarity: 'uncommon',
    activationType: 'active',
    compatibleClasses: ['cruiser', 'battlecruiser', 'battleship', 'dreadnought', 'carrier'],
    statModifiers: {
      shieldMultiplier: 1.1,
    },
    abilities: [
      {
        id: 'emergency_boost',
        name: 'Emergency Shield Boost',
        description: 'Instantly restore 50% shields and increase regen.',
        cooldown: 5,
        currentCooldown: 0,
        energyCost: 30,
        duration: 2,
        effect: {
          type: 'repair',
          target: 'self',
          repairAmount: 50,
          repairType: 'shield',
        } as AbilityEffect,
      },
    ],
    triggers: [],
    requirements: { techLevels: { shieldTech: 6 } },
    installCost: { metal: 6000, crystal: 8000, deuterium: 1500 },
    powerConsumption: 15,
    mass: 60,
  },

  // ========== ARMOR ==========
  reactive_armor: {
    name: 'Reactive Armor Plating',
    description: 'Armor that detonates outward to deflect incoming projectiles.',
    slot: 'armor',
    rarity: 'uncommon',
    activationType: 'passive',
    compatibleClasses: ['frigate', 'cruiser', 'battlecruiser', 'battleship', 'dreadnought', 'carrier', 'transport'],
    statModifiers: {
      armorMultiplier: 1.25,
      resistanceBonuses: {
        ballistic_resistance: 15,
        explosive_resistance: 10,
      },
    },
    abilities: [],
    triggers: [],
    requirements: { techLevels: { armorTech: 8 } },
    installCost: { metal: 8000, crystal: 3000, deuterium: 500 },
    powerConsumption: 5,
    mass: 150,
  },

  nanobot_hull: {
    name: 'Nanobot Hull Repair System',
    description: 'Self-repairing hull using nanobots.',
    slot: 'armor',
    rarity: 'rare',
    activationType: 'triggered',
    compatibleClasses: ['cruiser', 'battlecruiser', 'battleship', 'dreadnought', 'carrier'],
    statModifiers: {
      hullMultiplier: 1.1,
    },
    abilities: [],
    triggers: [
      {
        condition: { trigger: 'on_round_end', chance: 100 },
        effect: {
          type: 'repair',
          target: 'self',
          repairAmount: 5,
          repairType: 'hull',
        } as AbilityEffect,
      },
    ],
    requirements: { techLevels: { armorTech: 12 } },
    installCost: { metal: 12000, crystal: 8000, deuterium: 3000 },
    powerConsumption: 12,
    mass: 40,
  },

  // ========== ENGINES ==========
  afterburner: {
    name: 'Combat Afterburner',
    description: 'Emergency speed boost for evasion.',
    slot: 'engine',
    rarity: 'common',
    activationType: 'active',
    compatibleClasses: ['fighter', 'corvette', 'frigate', 'cruiser', 'battlecruiser'],
    statModifiers: {
      speedMultiplier: 1.1,
      evasionBonus: 5,
    },
    abilities: [
      {
        id: 'afterburn',
        name: 'Afterburn',
        description: 'Massively increase evasion for 2 rounds.',
        cooldown: 4,
        currentCooldown: 0,
        energyCost: 15,
        duration: 2,
        effect: {
          type: 'buff',
          target: 'self',
          modifiers: {
            evasionBonus: 30,
            speedMultiplier: 1.5,
          },
          duration: 2,
        } as AbilityEffect,
      },
    ],
    triggers: [],
    requirements: {},
    installCost: { metal: 3000, crystal: 2000, deuterium: 2000 },
    powerConsumption: 8,
    mass: 30,
  },

  // ========== COMPUTER SYSTEMS ==========
  targeting_computer: {
    name: 'Advanced Targeting Computer',
    description: 'Improves weapon accuracy and critical hit chance.',
    slot: 'computer',
    rarity: 'common',
    activationType: 'passive',
    compatibleClasses: ['corvette', 'frigate', 'cruiser', 'battlecruiser', 'battleship', 'dreadnought', 'carrier'],
    statModifiers: {
      accuracyBonus: 10,
      critChanceBonus: 5,
    },
    abilities: [],
    triggers: [],
    requirements: {},
    installCost: { metal: 2000, crystal: 5000, deuterium: 500 },
    powerConsumption: 8,
    mass: 20,
  },

  ecm_suite: {
    name: 'ECM Suite',
    description: 'Electronic countermeasures that disrupt enemy targeting.',
    slot: 'computer',
    rarity: 'uncommon',
    activationType: 'passive',
    compatibleClasses: ['frigate', 'cruiser', 'battlecruiser', 'battleship', 'dreadnought', 'carrier', 'utility'],
    statModifiers: {
      evasionBonus: 15,
      resistanceBonuses: {
        hack_defense: 20,
      },
    },
    abilities: [
      {
        id: 'jam',
        name: 'Jamming Pulse',
        description: 'Reduce enemy accuracy for 2 rounds.',
        cooldown: 4,
        currentCooldown: 0,
        energyCost: 25,
        duration: 2,
        effect: {
          type: 'debuff',
          target: 'all_enemies',
          statusEffect: 'sensors_jammed',
          duration: 2,
          strength: 20,
        } as AbilityEffect,
      },
    ],
    triggers: [],
    requirements: { techLevels: { computerTech: 8 } },
    installCost: { metal: 5000, crystal: 8000, deuterium: 1000 },
    powerConsumption: 12,
    mass: 25,
  },

  hacking_module: {
    name: 'Offensive Hacking Module',
    description: 'Enables cyber attacks against enemy systems.',
    slot: 'computer',
    rarity: 'rare',
    activationType: 'passive',
    compatibleClasses: ['frigate', 'cruiser', 'battlecruiser', 'carrier', 'utility'],
    statModifiers: {
      damageTypeBonuses: { hacking: 100 },
    },
    abilities: [
      {
        id: 'system_breach',
        name: 'System Breach',
        description: 'Attempt to disable an enemy ship\'s weapons.',
        cooldown: 5,
        currentCooldown: 0,
        energyCost: 40,
        duration: 0,
        effect: {
          type: 'debuff',
          target: 'enemy',
          statusEffect: 'weapons_disabled',
          duration: 2,
          strength: 100,
        } as AbilityEffect,
      },
    ],
    triggers: [],
    requirements: { techLevels: { computerTech: 12 } },
    installCost: { metal: 8000, crystal: 15000, deuterium: 2000 },
    powerConsumption: 18,
    mass: 30,
  },

  fire_control: {
    name: 'Integrated Fire Control',
    description: 'Coordinates all weapon systems for maximum efficiency.',
    slot: 'computer',
    rarity: 'rare',
    activationType: 'passive',
    compatibleClasses: ['battleship', 'dreadnought'],
    statModifiers: {
      damageMultiplier: 1.15,
      accuracyBonus: 15,
      critDamageBonus: 0.25,
    },
    abilities: [],
    triggers: [],
    requirements: { techLevels: { computerTech: 14 } },
    installCost: { metal: 12000, crystal: 18000, deuterium: 3000 },
    powerConsumption: 20,
    mass: 40,
  },

  // ========== SPECIAL ==========
  boarding_bay: {
    name: 'Assault Boarding Bay',
    description: 'Facility for launching boarding parties.',
    slot: 'special',
    rarity: 'uncommon',
    activationType: 'passive',
    compatibleClasses: ['cruiser', 'battlecruiser', 'battleship', 'dreadnought', 'carrier'],
    statModifiers: {
      boardingPowerBonus: 50,
      crewCapacityBonus: 20,
    },
    abilities: [],
    triggers: [],
    requirements: {},
    installCost: { metal: 10000, crystal: 5000, deuterium: 2000 },
    powerConsumption: 15,
    mass: 200,
  },

  point_defense_array: {
    name: 'Point Defense Array',
    description: 'Automated defense systems against missiles and fighters.',
    slot: 'special',
    rarity: 'common',
    activationType: 'passive',
    compatibleClasses: ['frigate', 'cruiser', 'battlecruiser', 'battleship', 'dreadnought', 'carrier'],
    statModifiers: {
      pointDefenseBonus: 30,
    },
    abilities: [],
    triggers: [],
    requirements: {},
    installCost: { metal: 4000, crystal: 3000, deuterium: 500 },
    powerConsumption: 10,
    mass: 50,
  },

  command_bridge: {
    name: 'Enhanced Command Bridge',
    description: 'Improved command and control for fleet coordination.',
    slot: 'special',
    rarity: 'rare',
    activationType: 'passive',
    compatibleClasses: ['battleship', 'dreadnought', 'carrier'],
    statModifiers: {
      accuracyBonus: 5,
      crewCombatBonus: 20,
    },
    abilities: [
      {
        id: 'rally',
        name: 'Rally Fleet',
        description: 'Boost nearby allies\' combat effectiveness.',
        cooldown: 6,
        currentCooldown: 0,
        energyCost: 50,
        duration: 3,
        effect: {
          type: 'buff',
          target: 'all_allies',
          modifiers: {
            damageMultiplier: 1.1,
            accuracyBonus: 10,
            critChanceBonus: 5,
          },
          duration: 3,
        } as AbilityEffect,
      },
    ],
    triggers: [],
    requirements: { techLevels: { computerTech: 10 } },
    installCost: { metal: 15000, crystal: 20000, deuterium: 5000 },
    powerConsumption: 25,
    mass: 100,
  },

  // ========== CONSUMABLES ==========
  repair_nanites: {
    name: 'Emergency Repair Nanites',
    description: 'One-time use emergency repair system.',
    slot: 'consumable',
    rarity: 'common',
    activationType: 'consumable',
    compatibleClasses: ['fighter', 'corvette', 'frigate', 'cruiser', 'battlecruiser', 'battleship', 'dreadnought', 'carrier', 'transport', 'utility'],
    statModifiers: {},
    abilities: [
      {
        id: 'emergency_repair',
        name: 'Emergency Repair',
        description: 'Instantly repair 30% of all damage.',
        cooldown: 0,
        currentCooldown: 0,
        energyCost: 0,
        duration: 0,
        effect: {
          type: 'repair',
          target: 'self',
          repairAmount: 30,
          repairType: 'all',
        } as AbilityEffect,
      },
    ],
    triggers: [],
    requirements: {},
    installCost: { metal: 1000, crystal: 2000, deuterium: 500 },
    powerConsumption: 0,
    mass: 5,
    usesRemaining: 1,
  },

  overcharge_cell: {
    name: 'Weapon Overcharge Cell',
    description: 'One-time massive damage boost.',
    slot: 'consumable',
    rarity: 'uncommon',
    activationType: 'consumable',
    compatibleClasses: ['fighter', 'corvette', 'frigate', 'cruiser', 'battlecruiser', 'battleship', 'dreadnought'],
    statModifiers: {},
    abilities: [
      {
        id: 'overcharge',
        name: 'Weapon Overcharge',
        description: 'Double weapon damage for 1 round.',
        cooldown: 0,
        currentCooldown: 0,
        energyCost: 0,
        duration: 1,
        effect: {
          type: 'buff',
          target: 'self',
          modifiers: {
            damageMultiplier: 2.0,
            critChanceBonus: 20,
          },
          duration: 1,
        } as AbilityEffect,
      },
    ],
    triggers: [],
    requirements: {},
    installCost: { metal: 2000, crystal: 3000, deuterium: 1000 },
    powerConsumption: 0,
    mass: 10,
    usesRemaining: 1,
  },
}

// ============================================================================
// TEMPLATE UTILITY FUNCTIONS
// ============================================================================

/**
 * Get all equipment templates of a specific slot type
 */
export function getTemplatesBySlot(slot: EquipmentSlot): Record<string, Omit<Equipment, 'id'>> {
  const result: Record<string, Omit<Equipment, 'id'>> = {}
  for (const [key, template] of Object.entries(EQUIPMENT_TEMPLATES)) {
    if (template.slot === slot) {
      result[key] = template
    }
  }
  return result
}

/**
 * Get all equipment templates of a specific rarity
 */
export function getTemplatesByRarity(rarity: Equipment['rarity']): Record<string, Omit<Equipment, 'id'>> {
  const result: Record<string, Omit<Equipment, 'id'>> = {}
  for (const [key, template] of Object.entries(EQUIPMENT_TEMPLATES)) {
    if (template.rarity === rarity) {
      result[key] = template
    }
  }
  return result
}

/**
 * Get all equipment templates compatible with a specific unit class
 */
export function getTemplatesForClass(unitClass: UnitClass): Record<string, Omit<Equipment, 'id'>> {
  const result: Record<string, Omit<Equipment, 'id'>> = {}
  for (const [key, template] of Object.entries(EQUIPMENT_TEMPLATES)) {
    if (template.compatibleClasses.includes(unitClass)) {
      result[key] = template
    }
  }
  return result
}

/**
 * Get template keys (IDs) for all equipment
 */
export function getTemplateKeys(): string[] {
  return Object.keys(EQUIPMENT_TEMPLATES)
}

File: unity_client/Assets/Scripts/SharedLib/Battle/EquipmentTemplates.cs
// =============================================================================
// EquipmentTemplates.cs — Ported from equipment-templates.ts
// Predefined equipment data for weapons, shields, armor, engines, etc.
// =============================================================================

using System.Collections.Generic;
using System.Linq;

namespace OGameX.SharedLib.Battle
{
    public static class EquipmentTemplates
    {
        private static readonly Dictionary<string, Equipment> _t;

        static EquipmentTemplates()
        {
            _t = new Dictionary<string, Equipment>();

            // --- WEAPONS ---
            Add("gauss_cannon", "Gauss Cannon", "Electromagnetic railgun with high armor penetration.",
                EquipmentSlot.Weapon, EquipmentRarity.Common, ActivationType.Passive,
                new[] { UnitClass.Frigate, UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought },
                new EquipmentStatModifiers { DamageTypeBonuses = new DamageTypes { Ballistic = 50 }, AccuracyBonus = 5 },
                5000, 2000, 500, 10, 100);

            Add("ion_disruptor", "Ion Disruptor", "Disrupts enemy shields and systems.",
                EquipmentSlot.Weapon, EquipmentRarity.Uncommon, ActivationType.Passive,
                new[] { UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought },
                new EquipmentStatModifiers { DamageTypeBonuses = new DamageTypes { Ionic = 80 } },
                8000, 6000, 1000, 15, 80, weaponsTech: 8,
                triggers: new[] { MkTrigger(TriggerType.OnHit, 20, AbilityEffectType.Debuff, StatusEffectType.ShieldDisruption, 2, 50) });

            Add("plasma_lance", "Plasma Lance", "High-energy plasma with devastating crits.",
                EquipmentSlot.Weapon, EquipmentRarity.Rare, ActivationType.Passive,
                new[] { UnitClass.Battleship, UnitClass.Dreadnought },
                new EquipmentStatModifiers { DamageTypeBonuses = new DamageTypes { Explosive = 100, Ballistic = 50 }, CritChanceBonus = 10, CritDamageBonus = 0.5 },
                15000, 10000, 3000, 25, 150, weaponsTech: 12);

            // --- SECONDARY ---
            Add("torpedo_launcher", "Torpedo Launcher", "Guided torpedoes with explosive damage.",
                EquipmentSlot.Secondary, EquipmentRarity.Common, ActivationType.Passive,
                new[] { UnitClass.Corvette, UnitClass.Frigate, UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought },
                new EquipmentStatModifiers { DamageTypeBonuses = new DamageTypes { Explosive = 75 } },
                4000, 2000, 1000, 8, 60);

            // --- SHIELDS ---
            Add("reinforced_shields", "Reinforced Shield Generator", "Enhanced shield capacity.",
                EquipmentSlot.Shield, EquipmentRarity.Common, ActivationType.Passive,
                new[] { UnitClass.Fighter, UnitClass.Corvette, UnitClass.Frigate, UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought, UnitClass.Carrier, UnitClass.Transport, UnitClass.Utility },
                new EquipmentStatModifiers { ShieldMultiplier = 1.2, ShieldRegenBonus = 5 },
                3000, 4000, 500, 10, 50);

            Add("adaptive_shields", "Adaptive Shield Matrix", "Shields that adapt to incoming damage.",
                EquipmentSlot.Shield, EquipmentRarity.Rare, ActivationType.Passive,
                new[] { UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought, UnitClass.Carrier },
                new EquipmentStatModifiers { ShieldMultiplier = 1.15, ResistanceBonuses = new ResistanceTypes { BallisticResistance = 10, IonicResistance = 10, ExplosiveResistance = 10 } },
                10000, 15000, 2000, 20, 80, shieldTech: 10);

            // --- ARMOR ---
            Add("reactive_armor", "Reactive Armor Plating", "Deflects incoming projectiles.",
                EquipmentSlot.Armor, EquipmentRarity.Uncommon, ActivationType.Passive,
                new[] { UnitClass.Frigate, UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought, UnitClass.Carrier, UnitClass.Transport },
                new EquipmentStatModifiers { ArmorMultiplier = 1.25, ResistanceBonuses = new ResistanceTypes { BallisticResistance = 15, ExplosiveResistance = 10 } },
                8000, 3000, 500, 5, 150, armorTech: 8);

            Add("nanobot_hull", "Nanobot Hull Repair", "Self-repairing hull nanobots.",
                EquipmentSlot.Armor, EquipmentRarity.Rare, ActivationType.Triggered,
                new[] { UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought, UnitClass.Carrier },
                new EquipmentStatModifiers { HullMultiplier = 1.1 },
                12000, 8000, 3000, 12, 40, armorTech: 12,
                triggers: new[] { MkTrigger(TriggerType.OnRoundEnd, 100, AbilityEffectType.Repair, repairAmt: 5, repairType: "hull") });

            // --- ENGINE ---
            Add("afterburner", "Combat Afterburner", "Emergency speed boost.",
                EquipmentSlot.Engine, EquipmentRarity.Common, ActivationType.Active,
                new[] { UnitClass.Fighter, UnitClass.Corvette, UnitClass.Frigate, UnitClass.Cruiser, UnitClass.Battlecruiser },
                new EquipmentStatModifiers { SpeedMultiplier = 1.1, EvasionBonus = 5 },
                3000, 2000, 2000, 8, 30);

            // --- COMPUTER ---
            Add("targeting_computer", "Advanced Targeting Computer", "Improved accuracy and crits.",
                EquipmentSlot.Computer, EquipmentRarity.Common, ActivationType.Passive,
                new[] { UnitClass.Corvette, UnitClass.Frigate, UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought, UnitClass.Carrier },
                new EquipmentStatModifiers { AccuracyBonus = 10, CritChanceBonus = 5 },
                2000, 5000, 500, 8, 20);

            Add("ecm_suite", "ECM Suite", "Electronic countermeasures.",
                EquipmentSlot.Computer, EquipmentRarity.Uncommon, ActivationType.Passive,
                new[] { UnitClass.Frigate, UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought, UnitClass.Carrier, UnitClass.Utility },
                new EquipmentStatModifiers { EvasionBonus = 15, ResistanceBonuses = new ResistanceTypes { HackDefense = 20 } },
                5000, 8000, 1000, 12, 25, computerTech: 8);

            Add("fire_control", "Integrated Fire Control", "Maximum weapon efficiency.",
                EquipmentSlot.Computer, EquipmentRarity.Rare, ActivationType.Passive,
                new[] { UnitClass.Battleship, UnitClass.Dreadnought },
                new EquipmentStatModifiers { DamageMultiplier = 1.15, AccuracyBonus = 15, CritDamageBonus = 0.25 },
                12000, 18000, 3000, 20, 40, computerTech: 14);

            // --- SPECIAL ---
            Add("boarding_bay", "Assault Boarding Bay", "Launch boarding parties.",
                EquipmentSlot.Special, EquipmentRarity.Uncommon, ActivationType.Passive,
                new[] { UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought, UnitClass.Carrier },
                new EquipmentStatModifiers { BoardingPowerBonus = 50, CrewCapacityBonus = 20 },
                10000, 5000, 2000, 15, 200);

            Add("point_defense_array", "Point Defense Array", "Anti-missile/fighter defense.",
                EquipmentSlot.Special, EquipmentRarity.Common, ActivationType.Passive,
                new[] { UnitClass.Frigate, UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought, UnitClass.Carrier },
                new EquipmentStatModifiers { PointDefenseBonus = 30 },
                4000, 3000, 500, 10, 50);

            Add("command_bridge", "Enhanced Command Bridge", "Fleet coordination bonuses.",
                EquipmentSlot.Special, EquipmentRarity.Rare, ActivationType.Passive,
                new[] { UnitClass.Battleship, UnitClass.Dreadnought, UnitClass.Carrier },
                new EquipmentStatModifiers { AccuracyBonus = 5, CrewCombatBonus = 20 },
                15000, 20000, 5000, 25, 100, computerTech: 10);

            // --- CONSUMABLES ---
            Add("repair_nanites", "Emergency Repair Nanites", "One-time 30% repair.",
                EquipmentSlot.Consumable, EquipmentRarity.Common, ActivationType.Consumable,
                new[] { UnitClass.Fighter, UnitClass.Corvette, UnitClass.Frigate, UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought, UnitClass.Carrier, UnitClass.Transport, UnitClass.Utility },
                new EquipmentStatModifiers(),
                1000, 2000, 500, 0, 5, uses: 1);

            Add("overcharge_cell", "Weapon Overcharge Cell", "One-time double damage.",
                EquipmentSlot.Consumable, EquipmentRarity.Uncommon, ActivationType.Consumable,
                new[] { UnitClass.Fighter, UnitClass.Corvette, UnitClass.Frigate, UnitClass.Cruiser, UnitClass.Battlecruiser, UnitClass.Battleship, UnitClass.Dreadnought },
                new EquipmentStatModifiers(),
                2000, 3000, 1000, 0, 10, uses: 1);
        }

        // --- Accessors ---
        public static Equipment Get(string id) => _t.GetValueOrDefault(id);
        public static IReadOnlyDictionary<string, Equipment> All => _t;
        public static Dictionary<string, Equipment> BySlot(EquipmentSlot s) =>
            _t.Where(kv => kv.Value.Slot == s).ToDictionary(kv => kv.Key, kv => kv.Value);
        public static Dictionary<string, Equipment> ByRarity(EquipmentRarity r) =>
            _t.Where(kv => kv.Value.Rarity == r).ToDictionary(kv => kv.Key, kv => kv.Value);
        public static Dictionary<string, Equipment> ForClass(UnitClass c) =>
            _t.Where(kv => kv.Value.CompatibleClasses.Contains(c)).ToDictionary(kv => kv.Key, kv => kv.Value);

        // --- Helpers ---
        private static void Add(string id, string name, string desc,
            EquipmentSlot slot, EquipmentRarity rarity, ActivationType act,
            UnitClass[] classes, EquipmentStatModifiers mods,
            int costM, int costC, int costD, int power, int mass,
            int weaponsTech = 0, int shieldTech = 0, int armorTech = 0, int computerTech = 0,
            EquipmentTrigger[] triggers = null, int? uses = null)
        {
            _t[id] = new Equipment
            {
                Id = id, Name = name, Description = desc,
                Slot = slot, Rarity = rarity, ActivationType = act,
                CompatibleClasses = new List<UnitClass>(classes),
                StatModifiers = mods,
                Triggers = triggers != null ? new List<EquipmentTrigger>(triggers) : new List<EquipmentTrigger>(),
                Requirements = new EquipmentRequirements { WeaponsTech = weaponsTech, ShieldTech = shieldTech, ArmorTech = armorTech, ComputerTech = computerTech },
                InstallCost = new UnitCost { Metal = costM, Crystal = costC, Deuterium = costD },
                PowerConsumption = power, Mass = mass, UsesRemaining = uses,
            };
        }

        private static EquipmentTrigger MkTrigger(TriggerType trigger, double chance,
            AbilityEffectType effectType,
            StatusEffectType status = default, int duration = 0, double strength = 0,
            double repairAmt = 0, string repairType = null)
        {
            return new EquipmentTrigger
            {
                Condition = new TriggerCondition { Trigger = trigger, Chance = chance },
                Effect = new AbilityEffect
                {
                    Type = effectType, Target = "self",
                    StatusEffect = status, Duration = duration, Strength = strength,
                    RepairAmount = repairAmt, RepairType = repairType,
                },
            };
        }
    }
}
